Add documentation and a unit test for returning multiple arrays in a smocked function

When trying to mock an UniswapV3Pool, to be accessed through the Oracle library, we're pointing to the `observe(secondsAgo)` function.

Code to be used in our contract:
``` (int56[] memory tickCumulatives, uint160[] memory secondsPerLiquidityCumulativeX128s) = IUniswapV3Pool(pool).observe(secondsAgos);```

UniswapV3Pool.sol :

```
Oracle.Observation[65535] public override observations;
    [...]
    /// LINE 236
    function observe(uint32[] calldata secondsAgos)
        external
        view
        override
        noDelegateCall
        returns (int56[] memory tickCumulatives, uint160[] memory secondsPerLiquidityCumulativeX128s)
    {  ...  }
```


When trying to smock the answer of the address, have tried multiple options:
`pool.smocked.observe.will.return.with([ 1, 1 ])`
`pool.smocked.observe.will.return.with([ [1], [1] ])`
```pool.smocked.observe.will.return.with({
'tickCumulatives':1,
'secondsPerLiquidityCumulativeX128s':1
})
```
```
pool.smocked.observe.will.return.with({
'tickCumulatives':[1,2,3],
'secondsPerLiquidityCumulativeX128s':[1,2,3]
})
```

I've tried also replacing the numbers with BigNumber.from(0) but still, every try I'm getting the following error `Error: Transaction reverted: function returned an unexpected amount of data`

`return.with([ 1234 ], [ 5678 ])` doesn't seem to be compilable with typescript:
```
TSError: ⨯ Unable to compile TypeScript:
test/unit/UniV3PoolTest.spec.ts:27:50 - error TS2554: Expected 0-1 arguments, but got 2.
27     pool.smocked.observe.will.return.with([1234],[5678]);
```

I'm using:
```json
    "hardhat": "2.4.1",
    "@typechain/hardhat": "2.1.1",
    "@eth-optimism/smock": "1.1.6",
```

--- a/.changeset/wise-mails-laugh.md
+++ b/.changeset/wise-mails-laugh.md
@@ -1,0 +1,5 @@
+---
+'@eth-optimism/smock': patch
+---
+
+Add a test and a doc section for returning multiple uint256 arrays

--- a/packages/smock/README.md
+++ b/packages/smock/README.md
@@ -240,6 +240,25 @@ MyMockContract.smocked.myFunction.will.return.with({
 console.log(await MyMockContract.myFunction()) // ['Some value', 1234, true]
 ```
 
+### Returning a Multiple Arrays
+```typescript
+import { ethers } from 'hardhat'
+import { smockit } from '@eth-optimism/smock'
+
+const MyContractFactory = await ethers.getContractFactory('MyContract')
+const MyContract = await MyContractFactory.deploy(...)
+
+// Smockit!
+const MyMockContract = await smockit(MyContract)
+
+MyMockContract.smocked.myFunction.will.return.with([
+  [1234, 5678],
+  [4321, 8765]
+])
+
+console.log(await MyMockContract.myFunction()) // [ [1234, 5678], [4321, 8765] ]
+```
+
 ### Returning a Function
 ```typescript
 import { ethers } from 'hardhat'

--- a/packages/smock/test/contracts/TestHelpers_BasicReturnContract.sol
+++ b/packages/smock/test/contracts/TestHelpers_BasicReturnContract.sol
@@ -132,6 +132,14 @@ contract TestHelpers_BasicReturnContract {
         )
     {}
 
+    function getMultipleUint256Arrays()
+        public
+        returns (
+            uint256[] memory,
+            uint256[] memory
+        )
+    {}
+
     function overloadedFunction(
         uint256 _paramA,
         uint256 _paramB

--- a/packages/smock/test/smockit/function-manipulation.spec.ts
+++ b/packages/smock/test/smockit/function-manipulation.spec.ts
@@ -533,6 +533,25 @@ describe('[smock]: function manipulation tests', () => {
               expect(result[i]).to.deep.equal(expected[i])
             }
           })
+
+          it('should be able to return multiple arrays of uint256 values', async () => {
+            const expected = [
+              [1234, 2345, 3456, 4567, 5678, 6789].map((n) => {
+                return BigNumber.from(n)
+              }),
+              [1234, 2345, 3456, 4567, 5678, 6789].map((n) => {
+                return BigNumber.from(n)
+              }),
+            ]
+            mock.smocked.getMultipleUint256Arrays.will.return.with(expected)
+
+            const result = await mock.callStatic.getMultipleUint256Arrays()
+            for (let i = 0; i < result.length; i++) {
+              for (let j = 0; j < result[i].length; j++) {
+                expect(result[i][j]).to.deep.equal(expected[i][j])
+              }
+            }
+          })
         })
       })
     })

